src/plugins/layout/shadowdom.ts

```diff
@@ -240,22 +187,11 @@ export class ShadowDom {
     };
 
     // Collect all new nodes in the top-down order
-    let topNewNodes = this.doc.getElementsByClassName(NewNodeClassName);
-    while (topNewNodes.length > 0) {
-      let topNode = topNewNodes[0] as IShadowDomNode;
-      let discoverQueue: IShadowDomNode[] = [ topNode ];
-      while (discoverQueue.length > 0) {
-        let next = discoverQueue.shift();
-        if (this.hasClass(next, NewNodeClassName)) {
-          summary.newNodes.push(next);
-          this.removeClass(next, NewNodeClassName);
-          // Add children right-to-left
-          let children = next.childNodes;
-          for (let i = children.length - 1; i >= 0; i--) {
-            discoverQueue.push(children[i] as IShadowDomNode);
-          }
-        }
-      }
+    let newNodes = this.doc.getElementsByClassName(NewNodeClassName);
+    while (newNodes.length > 0) {
```

Comment:
newNodes is a 'live' HTMLCollection of nodes with the requested className. When I remove class from a node, that node automatically disappears from that collection.

---
In reply to: [131238745](https://github.com/Microsoft/clarity-js/pull/25#discussion_r131238745) [](ancestors = 131238745)